Enforce adoption of Dispatched(From|To) across message receivers

gavin-apple opened the pull request:
#### c6389c6268ff2ccb00c3ee1479fefb10e38b8405
<pre>
Enforce adoption of Dispatched(From|To) across message receivers
<a href="https://bugs.webkit.org/show_bug.cgi?id=294835">https://bugs.webkit.org/show_bug.cgi?id=294835</a>
<a href="https://rdar.apple.com/problem/154096608">rdar://problem/154096608</a>

Reviewed by Alex Christensen.

This PR enforces adoption of Dispatched(From|To) across our IPC receivers.
It introduces a pair of new ExceptionForDispatched(From|To) which can be
used in places where we still rely on multiple processes interacting
with a single receiver. The intention is to burn these down for all but
explcitly shared receivers.

* Source/WebKit/GPUProcess/RemoteSharedResourceCache.messages.in:
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.messages.in:
* Source/WebKit/Scripts/generate-derived-log-sources.py:
(generate_messages_file):
* Source/WebKit/Scripts/webkit/parser.py:
(parse):
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBuffer.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithEnabledBy.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithEnabledByAndConjunction.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithEnabledByOrConjunction.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithIfMessage.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithIfReceiver.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithImageData.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiver.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphore.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithSpanOfConst.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithStream.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBatched.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBuffer.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamServerConnectionHandle.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclass.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassAndWantsAsyncDispatch.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassAndWantsDispatch.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithWantsAsyncDispatch.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithWantsDispatch.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithWantsDispatchNoSyncMessages.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributes.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithoutUsingIPCConnection.messages.in:
* Source/WebKit/Shared/API/Cocoa/RemoteObjectRegistry.messages.in:
* Source/WebKit/Shared/ApplePay/WebPaymentCoordinatorProxy.messages.in:
* Source/WebKit/Shared/AuxiliaryProcess.messages.in:
* Source/WebKit/Shared/IPCConnectionTester.messages.in:
* Source/WebKit/Shared/IPCStreamTester.messages.in:
* Source/WebKit/Shared/IPCStreamTesterProxy.messages.in:
* Source/WebKit/Shared/IPCTester.messages.in:
* Source/WebKit/Shared/IPCTesterReceiver.messages.in:
* Source/WebKit/Shared/LogStream.messages.in:
* Source/WebKit/Shared/Notifications/NotificationManagerMessageHandler.messages.in:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.messages.in:
* Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.messages.in:
* Source/WebKit/UIProcess/dmabuf/AcceleratedBackingStoreDMABuf.messages.in:
* Source/WebKit/WebProcess/ApplePay/WebPaymentCoordinator.messages.in:
* Source/WebKit/WebProcess/GPU/media/ios/RemoteMediaSessionHelper.messages.in:
* Source/WebKit/WebProcess/Geolocation/WebGeolocationManager.messages.in:
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.messages.in:
* Source/WebKit/WebProcess/Notifications/WebNotificationManager.messages.in:
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.messages.in:
* Source/WebKit/WebProcess/Storage/WebSharedWorkerContextManagerConnection.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebSpeechRecognitionConnection.messages.in:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.messages.in:
* Source/WebKit/WebProcess/cocoa/RemoteCaptureSampleManager.messages.in:
* Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.messages.in:
* Source/WebKit/WebProcess/glib/SystemSettingsManager.messages.in:
* Source/WebKit/WebProcess/glib/UserMediaCaptureManager.messages.in:
* Source/WebKit/webpushd/PushClientConnection.messages.in:

Canonical link: <a href="https://commits.webkit.org/298555@main">https://commits.webkit.org/298555@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a01a4a48be71d49c98ef724aff73966d532b6f48

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115751 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35414 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25937 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121799 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66286 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e1638480-9f06-4997-9cfb-55c2b59600d5) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117640 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36095 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43997 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87924 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42652 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/90d2dbcf-0ad2-439a-933f-d6ad3f9d6c50) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118699 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28794 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103867 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68325 "Passed tests") | | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/115125 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27933 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21982 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65470 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98182 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22103 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124949 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42646 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31980 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96682 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43013 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100056 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96468 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24566 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41734 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19587 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38579 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42537 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/48109 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42010 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45340 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43718 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ac5bf1b81f84a44f439c4f31f02c5c8cf84039de. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47054

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/979fe32ee346c912b232cd47bf7370cb10077aa8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47054

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/39015fa0477f1d1703bd2384715ee780adea747c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47054

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b60bd4ef4a8705fec8e0587cad0e9fff22bb5fa9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47054

webkit-commit-queue commented:
Commit message contains (OOPS!), blocking PR #47054. Details: [Build #24523](https://ews-build.webkit.org/#/builders/19/builds/24523)

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9be8b937185d6b77f44e29a001069c15849fb612. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47054

gavin-apple committed:
Enforce adoption of Dispatched(From|To) across message receivers
https://bugs.webkit.org/show_bug.cgi?id=294835
rdar://problem/154096608

Reviewed by Alex Christensen.

This PR enforces adoption of Dispatched(From|To) across our IPC receivers.
It introduces a pair of new ExceptionForDispatched(From|To) which can be
used in places where we still rely on multiple processes interacting
with a single receiver. The intention is to burn these down for all but
explcitly shared receivers.

* Source/WebKit/GPUProcess/RemoteSharedResourceCache.messages.in:
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.messages.in:
* Source/WebKit/Scripts/generate-derived-log-sources.py:
(generate_messages_file):
* Source/WebKit/Scripts/webkit/parser.py:
(parse):
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBuffer.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithEnabledBy.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithEnabledByAndConjunction.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithEnabledByOrConjunction.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithIfMessage.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithIfReceiver.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithImageData.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiver.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphore.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithSpanOfConst.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithStream.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBatched.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBuffer.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamServerConnectionHandle.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclass.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassAndWantsAsyncDispatch.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassAndWantsDispatch.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithWantsAsyncDispatch.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithWantsDispatch.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithWantsDispatchNoSyncMessages.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributes.messages.in:
* Source/WebKit/Scripts/webkit/tests/TestWithoutUsingIPCConnection.messages.in:
* Source/WebKit/Shared/API/Cocoa/RemoteObjectRegistry.messages.in:
* Source/WebKit/Shared/ApplePay/WebPaymentCoordinatorProxy.messages.in:
* Source/WebKit/Shared/AuxiliaryProcess.messages.in:
* Source/WebKit/Shared/IPCConnectionTester.messages.in:
* Source/WebKit/Shared/IPCStreamTester.messages.in:
* Source/WebKit/Shared/IPCStreamTesterProxy.messages.in:
* Source/WebKit/Shared/IPCTester.messages.in:
* Source/WebKit/Shared/IPCTesterReceiver.messages.in:
* Source/WebKit/Shared/LogStream.messages.in:
* Source/WebKit/Shared/Notifications/NotificationManagerMessageHandler.messages.in:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.messages.in:
* Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.messages.in:
* Source/WebKit/UIProcess/dmabuf/AcceleratedBackingStoreDMABuf.messages.in:
* Source/WebKit/WebProcess/ApplePay/WebPaymentCoordinator.messages.in:
* Source/WebKit/WebProcess/GPU/media/ios/RemoteMediaSessionHelper.messages.in:
* Source/WebKit/WebProcess/Geolocation/WebGeolocationManager.messages.in:
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.messages.in:
* Source/WebKit/WebProcess/Notifications/WebNotificationManager.messages.in:
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.messages.in:
* Source/WebKit/WebProcess/Storage/WebSharedWorkerContextManagerConnection.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebSpeechRecognitionConnection.messages.in:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.messages.in:
* Source/WebKit/WebProcess/cocoa/RemoteCaptureSampleManager.messages.in:
* Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.messages.in:
* Source/WebKit/WebProcess/glib/SystemSettingsManager.messages.in:
* Source/WebKit/WebProcess/glib/UserMediaCaptureManager.messages.in:
* Source/WebKit/webpushd/PushClientConnection.messages.in:

Canonical link: https://commits.webkit.org/298555@main
